bin/install

```diff
@@ -0,0 +1,190 @@
+
+## Start Helpers
+shell_join() {
+  local arg
+  printf "%s" "$1"
+  shift
+  for arg in "$@"
+  do
+    printf " "
+    printf "%s" "${arg// /\ }"
+  done
+}
+
+ohai() {
+  printf "${Purple}==>${Bold} %s${Off}\n" "$(shell_join "$@")"
+}
+
+execute() {
+  if ! "$@"
+  then
+    abort "$(printf "Failed during: %s" "$(shell_join "$@")")"
+  fi
+}
+## End Helpers
+
+# Set Up Start
+ohai "Configure Network Access"
+
+message=$(cat <<-END
+
+    By default, Terrastories is setup with network access at terrastories.local.
+    Would you like to change your host name to access it at a custom location?
+
+        (1) Yes, please.
+        (2) No, default is fine.
+
+    Your choice: [2] 
+END
+)
+
+hostname=$(grep "^HOST_HOSTNAME=.*" .env.offline | cut -s -d= -f2 -)
+if [ ! -z $hostname ]; then
+  echo "Terrastories is currently configured with network access at $hostname."
+  echo
+else
+  read -p "$message" addhosts
+
+  hostname=$(hostname)
+  if [ "$addhosts" = "1" ]; then
+    echo
+    read -p "Specify hostname: [$hostname] " hosts
+    hostname=${hosts:-$hostname}
+  fi
+  # Enable HOST_HOSTNAME
+  if [ ! -z "$hostname" ]; then
+    grep -v "HOST_HOSTNAME" .env.offline > .envnew; mv .envnew .env.offline
+    echo "HOST_HOSTNAME=$hostname" >> .env.offline
+  fi
+
+  echo -e "...${Green}done!${Off}\n"
+fi
+
+ohai Configure Host Domain
+# Add to /etc/hosts if not already there
+if [ "$hostname" = "$(hostname)" ]; then
+  echo "Machine hostname configured by default"
+  echo
+elif ! grep -q "$hostname" /etc/hosts; then
+  read -p "$(
+    cat <<MODHOSTS
+
+In order to utilize $hostname, you must add it to your /etc/hosts file.
+This can be done manually by editing your /etc/hosts file, or you can provide your
+password when prompted to allow us to automatically configure them.
+
+  Choose one:
+    (1) Manual
+    (2) Automatic
+
+Your choice: [2] 
+MODHOSTS
+  )" modHosts
+
+  if [ "$modHosts" = "1" ]; then
+    read -p "$(
+  cat <<HOSTS
+
+In a new terminal window, open /etc/hosts with appropriate write permissions
+and add the following line:
+
+      127.0.0.1 $hostname
+
+Press any key to continue or q to quit. 
+HOSTS
+    )" -n 1 k
+    if [[ "$k" = "q" ]]; then
+      exit
+    fi
+  else
+    echo "Automatically configuring terrastories.local. You may prompted for your password."
+    sudo -- sh -c "echo '# Added by Terrastories\n127.0.0.1 $hostname\n127.0.0.1 terrastories.local\n# End of section' >> /etc/hosts"
+    echo -e "...${Green}done!${Off}"
+  fi
+fi
+
+echo
+ohai Configuring Terrastories default map package…
+# Download default .pmtiles
+if [ ! -f map/terrastories-map/tiles.pmtiles ]; then
+  curl -fsSL "https://t.ly/OTZpR" -o "map/terrastories-map/tiles.pmtiles" --create-dirs
+fi
+curl -fsSL "https://github.com/protomaps/basemaps-assets/archive/refs/heads/main.tar.gz" | tar xzf - -C map/terrastories-map --strip-components=1 basemaps-assets-main/fonts basemaps-assets-main/sprites
```

Comment:
I purposely included this setup.

1. It's a good middle ground for deprecation and moving to the supported offline setup that uses PM tiles.
2. Just because they are installing with an existing Tileserver configuration doesn't exclude them from using pmtiles. The config allows for swapping between the two.
3. This will become even more relevant as we add additional functionality that will allow us to convert their mbtiles and styles into pmtiles and styles.

Reply:
I think it's potentially confusing for most of our offline users who I don't anticipate will want to swap between two different map tile formats, but it's fine to keep it. And any nudging towards pmtiles and notification about the eventual deprecation of tileserver / mbtiles is a good thing.